akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -941,35 +940,69 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
   /**
    * Initiates a new round of gossip.
    */
-  def gossip(): Unit = {
-
+  def gossip(): Unit =
     if (!isSingletonCluster) {
-      val localGossip = latestGossip
+      val gossipTargets =
+        // TODO break this two methods out and unit test?
+        if (latestGossip.isMultiDc) multiDcGossipTargets()
+        else localDcGossipTargets()
+
+      selectRandomNode(gossipTargets) match {
+        case Some(peer) ⇒ gossipTo(peer)
+        case None       ⇒ // nothing to see here
+      }
+    }
 
-      val preferredGossipTargets: Vector[UniqueAddress] =
-        if (ThreadLocalRandom.current.nextDouble() < adjustedGossipDifferentViewProbability) {
-          // If it's time to try to gossip to some nodes with a different view
-          // gossip to a random alive member with preference to a member with older gossip version
-          localGossip.members.collect {
-            case m if !localGossip.seenByNode(m.uniqueAddress) && validNodeForGossip(m.uniqueAddress) ⇒
-              m.uniqueAddress
-          }(breakOut)
-        } else Vector.empty
-
-      if (preferredGossipTargets.nonEmpty) {
-        val peer = selectRandomNode(preferredGossipTargets)
-        // send full gossip because it has different view
-        peer foreach gossipTo
-      } else {
-        // Fall back to localGossip; important to not accidentally use `map` of the SortedSet, since the original order is not preserved)
-        val peer = selectRandomNode(localGossip.members.toIndexedSeq.collect {
-          case m if validNodeForGossip(m.uniqueAddress) ⇒ m.uniqueAddress
-        })
-        peer foreach { node ⇒
-          if (localGossip.seenByNode(node)) gossipStatusTo(node)
```

Comment:
Doing the `seenByNode` twice shouldn't be a problem, it's just a Map lookup.

For the cross DC it's much more likely that the target anyway has the latest state even though it's not marked as such in the seen yet, and therefore I think it makes sense to always use the `gossipStatusTo` to avoid transferring the full thing unless necessary.